sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala

```diff
@@ -807,6 +807,8 @@ class JDBCSuite extends QueryTest with SharedSparkSession {
       assert(doCompileFilter(LessThan(col0, 5)) === """"col0" < 5""")
       assert(doCompileFilter(LessThan(col0,
         Timestamp.valueOf("1995-11-21 00:00:00.0"))) === """"col0" < '1995-11-21 00:00:00.0'""")
+      assert(doCompileFilter(LessThan(col0,
+        LocalDateTime.parse("2007-12-03T10:15:30"))) === """"col0" < '2007-12-03T10:15:30'""")
```

Comment:
Shouldn't be consistent to the above, and don't have `T` between date and time?

Reply:
Here `LocalDateTime.parse("2007-12-03T10:15:30")` is the default format for the parser and fail without the 'T'. I have change it to other constructor to avoid confusion